src/main/java/org/veupathdb/service/eda/compute/plugins/correlation/CorrelationPlugin.java

```diff
@@ -0,0 +1,114 @@
+      );
+  }
+
+  @Override
+  protected void execute() {
+
+    CorrelationComputeConfig computeConfig = getConfig();
+    PluginUtil util = getUtil();
+    ReferenceMetadata meta = getContext().getReferenceMetadata();
+
+    String entityId = computeConfig.getCollectionVariable().getEntityId();
+    EntityDef entity = meta.getEntity(entityId).orElseThrow();
+    VariableDef computeEntityIdVarSpec = util.getEntityIdVarSpec(entityId);
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    String method = computeConfig.getCorrelationMethod().getValue();
+
+    // Get record id columns
+    List<VariableDef> idColumns = new ArrayList<>();
+    for (EntityDef ancestor : meta.getAncestors(entity)) {
+      idColumns.add(ancestor.getIdColumnDef());
+    }
+
+    HashMap<String, InputStream> dataStream = new HashMap<>();
+    dataStream.put(INPUT_DATA, getWorkspace().openStream(INPUT_DATA));
+    
+    RServe.useRConnectionWithRemoteFiles(dataStream, connection -> {
+      connection.voidEval("print('starting correlation computation')");
+
+      // Read in the abundance data
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      computeInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+      connection.voidEval("abundanceData <- " + INPUT_DATA); // Renaming here so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      // TO DO get only the continuous sample metadata!!
+      // List<VariableSpec> sampleMetadataVars = ListBuilder.asList(comparisonVariableSpec);
+      // sampleMetadataVars.add(computeEntityIdVarSpec);
+      // connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, sampleMetadataVars));
+      // connection.voidEval("sampleMetadata <- " + INPUT_DATA);
+
+
+      // Turn the list of id columns into an array of strings for R
+      List<String> dotNotatedIdColumns = idColumns.stream().map(VariableDef::toDotNotation).toList();
+      String dotNotatedIdColumnsString = "c(";
+      boolean first = true;
+      for (String idCol : dotNotatedIdColumns) {
+        if (first) {
+          first = false;
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + singleQuote(idCol);
+        } else {
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + "," + singleQuote(idCol);
+        }
+      }
+      dotNotatedIdColumnsString = dotNotatedIdColumnsString + ")";
+
+
+      // Set up input assay data.
+      // TEMP until we have the continuous sample metadata, we're letting the abundance data sub for continuous sample metadata
+      connection.voidEval("inputData <- microbiomeComputations::AbundanceData(data=abundanceData" + 
+                                                                          ", sampleMetadata=abundanceData" +
```

Comment:
abundanceData is both data and sampleMetadata?

Reply:
right. Just for now becasue we dont have that nice grabVariablesByPredicate function yet. I'm trying to get something working and sending worthwhile data to the frontend. I think the frontend will be pretty tricky so i want to get them data as asap as possible.
So we're just faking data again temporarily until we can get all the metadata we need.